app/models/target.rb

```diff
@@ -19,24 +19,38 @@
 #
 
 class Target < ApplicationRecord
-  MAXIMUM_NUMBER_OF_TARGETS_PER_USER = 10
+  MAX_TARGETS_PER_USER = 10
 
   belongs_to :topic
   belongs_to :user
 
+  acts_as_mappable default_units: :kms,
+                   default_formula: :flat,
+                   distance_field_name: :distance,
+                   lat_column_name: :latitude,
+                   lng_column_name: :longitude
+
   validates :title, presence: true, uniqueness: true
   validates :latitude, :longitude, presence: true
 
   validate :target_limit_per_user
 
+  def search_compatible_targets
+    compatible_targets = []
+    Target.where.not(user_id: user_id).where(topic_id: topic_id).each do |target|
```

Comment:
I'll suggest creating two scopes on Target like `scope :all_except_mine, ->(user_id) { where.not(user_id: user_id) }` & `scope :with_topic, (topic_id) { where(topic_id: topic_id) }` and then call it here like `Target.all_except_mine(user_id).with_topic(topic_id)`

Reply:
also as you are already in the Target model, no need to call `Target`

Reply:
You can apply the same logic as in my previous comment, using map